### DISCUSSION TITLE
Using S3 storage vs local files impact cpu resources?

### DISCUSSION THREAD

**Discussion by caonima212** [pocketbase/pocketbase#4545] [Q&A]
Is using s3 storage better for big file upload? And what about the cpu resources during read/write process? does it impact on my server resources too?

**Comment by ganigeorgiev** [comment]
I don't think the S3 upload will have much impact on the CPU consumption, but S3 file upload/access will be definitely slower (_especially if the S3 server is in a different location_) and more memory intensive compared to direct interaction with the local filesystem.

You'll have to benchmark it on your own if you need specific numbers as it will depend on too many factors - the S3 provider and its used hardware, rate/throttling limits, the network transfer speed between the PocketBase server and the S3, etc.

In general, if your server has enough disk space to store the uploaded files and there are no expectations to grow beyond that (_note that some VPS providers offer also extra "Volumes" or similar network block storage options_), prefer using the local filesystem, otherwise - S3.

Also keep in mind that I haven't done detailed profiling yet and PocketBase may not be optimized for uploading/serving large files (ex. 500MB+) at the moment. There is also no support out of the box for chunked or partial uploads (aka. no pause/resume).
